framework/core/js/src/common/utils/string.ts

```diff
@@ -16,7 +16,8 @@ export function truncate(string: string, length: number, start: number = 0): str
 export function slug(string: string): string {
```

Comment:
I think I'd prefer to do this by taking in a number of slugging "modes" for strictness, maybe represented with an enum / match?